libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2222,24 +2247,20 @@ string YulUtilFunctions::readFromStorageReferenceType(Type const& _type)
 	for (size_t i = 0; i < structMembers.size(); ++i)
 	{
 		auto const& [memberSlotDiff, memberStorageOffset] = structType.storageOffsetsOfMember(structMembers[i].name);
+		solAssert(structMembers[i].type->isValueType() || memberStorageOffset == 0, "");
```

Comment:
Added assertion, to have it here as well. Probably it is done in many other places, but it was missing here